tools/encryption/encryptionutils/encryption-helper.go

```diff
@@ -0,0 +1,243 @@
+// Helper functions for the SwG Encryption Script.
+package encryptionutils
+
+import (
+	"bytes"
+    "encoding/base64"
+	"fmt"
+	"io"
+	"golang.org/x/net/html"
+	"golang.org/x/net/html/atom"
+	"github.com/golang/protobuf/proto"
+	"github.com/google/tink/go/aead"
+	"github.com/google/tink/go/hybrid"
+	"github.com/google/tink/go/insecurecleartextkeyset"
+	"github.com/google/tink/go/keyset"
+	"github.com/google/tink/go/core/registry"
+	gcmpb "github.com/google/tink/proto/aes_gcm_go_proto"
+	tinkpb "github.com/google/tink/proto/tink_go_proto"
+	"net/http"
+	"strings"
+)
+
+const AES_GCM_KEY_URL string = "type.googleapis.com/google.crypto.tink.AesGcmKey" 
+const AES_GCM_KEY_SIZE uint32 = 16
+
+// Public function to generate an encrypted HTML document given the original.
+func GenerateEncryptedDocument(html_str string, public_key_url string, access_requirement string) (string, error) {
```

Comment:
This method is stateless. Could it be more stateful so it can encrypt multiple documents? Seems reasonable that a user would want to encrypt more than one?

By state, I mean it can save the Google Public Key (so it's not re-retrieving it over and over). I'm not sure if the AesKey has to be regenerated per doc?

Reply:
I modified GenerateEncryptedDocument so that it accepts a public key as input, which means that I can call GenerateEncryptedDocument multiple times with the same key in the script.

Reply:
To confirm, the AesKey has to be generated per document.